plugin/storage/cassandra/cassandra3v001-schema.sh

```diff
@@ -0,0 +1,191 @@
+#!/bin/bash
+
+# Notes:
+# gc_grace_seconds is non-zero, see: http://www.uberobert.com/cassandra_gc_grace_disables_hinted_handoff/
+# For TTL of 2 days, compaction window is 1 hour, rule of thumb here: http://thelastpickle.com/blog/2016/12/08/TWCS-part1.html
+function usage {
+    >&2 echo "Error: $1"
+    >&2 echo "Usage: $0 {prod|test} [datacenter] | cqlsh"
+    >&2 echo ""
+    >&2 echo "Other parameters can be set via environment:"
+    >&2 echo "  TTL - default time to live, in seconds (default: 172800, 2 days)"
+    >&2 echo "  KEYSPACE - keyspace (default: jaeger_v1_$datacenter)"
+    >&2 echo "  REPLICATION_FACTOR - replication factor for prod (default: 2)"
+    exit 1
+}
+
+default_time_to_live=${TTL:-172800}
+replication_factor=${REPLICATION_FACTOR:-2}
+
+if [[ "$1" == "" ]]; then
+    usage "missing environment"
+elif [[ "$1" == "prod" ]]; then
+    if [[ "$2" == "" ]]; then usage "missing datacenter"; fi
+    datacenter=$2
+    replication="{'class': 'NetworkTopologyStrategy', '$datacenter': '${replication_factor}' }"
+elif [[ "$1" == "test" ]]; then 
+    datacenter=${2:-'test'}
+    replication="{'class': 'SimpleStrategy', 'replication_factor': '1'}"
+else
+    usage "invalid environment $1, expecting prod or test"
+fi
+
+keyspace=${KEYSPACE:-"jaeger_v1_${datacenter}"}
+
+cat - <<EOF
+CREATE KEYSPACE IF NOT EXISTS ${keyspace} WITH replication = ${replication};
+
+CREATE TYPE IF NOT EXISTS ${keyspace}.keyvalue (
+    key             text,
+    value_type      text,
+    value_string    text,
+    value_bool      boolean,
+    value_long      bigint,
+    value_double    double,
+    value_binary    blob,
+);
+
+CREATE TYPE IF NOT EXISTS ${keyspace}.log (
+    ts      bigint,
+    fields  list<frozen<keyvalue>>,
+);
+
+CREATE TYPE IF NOT EXISTS ${keyspace}.span_ref (
+    ref_type        text,
+    trace_id        blob,
+    span_id         bigint,
+);
+
+CREATE TYPE IF NOT EXISTS ${keyspace}.process (
+    service_name    text,
+    tags            list<frozen<keyvalue>>,
+);
+
+CREATE TABLE IF NOT EXISTS ${keyspace}.service_names (
```

Comment:
can we move these two tables _after_ `traces`? Better to keep UDTs next to the table that uses them